ui/src/routes/__root.tsx

```diff
@@ -18,15 +18,26 @@
  */
 
 import { createRootRouteWithContext, Outlet } from '@tanstack/react-router';
-import { TanStackRouterDevtools } from '@tanstack/router-devtools';
+import React, { Suspense } from 'react';
 
 import { RouterContext } from '@/app';
 
+// Don't use Router devtools in production.
+const TanStackRouterDevtools = import.meta.env.PROD
+  ? () => null
+  : React.lazy(() =>
+      import('@tanstack/router-devtools').then((res) => ({
+        default: res.TanStackRouterDevtools,
+      }))
+    );
```

Comment:
Crazy how complicated this is 🤯

Reply:
It could done simpler by just not rendering the devtools if the environment is production, but that way the component would still get bundled into the production bundle. This guard saves probably a few kB on each load. :D